mbuild/box.py

```diff
@@ -2,151 +2,333 @@
+    2. Vectors are in a right-handed basis (determinant of matrix is +)
+    3. The first vector is aligned along the [1,0,0] direction
+    4. The second vector in aligned along the xy plane
+    5. The third vector can align freely in the x,y, and +z direction
+
+    If the vectors are not right-handed, a warning will be raised,
+    and the vectors will be transformed into the right-handed coordinate
+    system.
+
+    If the three vectors are not following conventions 3-5, the matrix will
+    be transformed to comply with them, and will also raise a warning.
     """
-    def __new__(cls, array, var=None, box=None, dtype=np.float):
-        _array = np.asarray(array, dtype).view(cls)
-        _array.var = var
-        _array.box = box
-        return _array
-
-    def __setitem__(self, key, val):
-        array = list(self)
-        array[key] = val
-        if self.var == "maxs":
-            msg = "Given max value is less than box's min value" 
-            assert val >= self.box.mins[key], msg
-            self.box.maxs = array
-        elif self.var == "mins":
-            msg = "Given min value is more than box's max value"
-            assert val <= self.box.maxs[key], msg
-            self.box.mins = array
-        elif self.var == "lengths":
-            msg = "Given length value is negative"
-            assert val >= 0, msg
-            self.box.lengths = array
-        else:
-            self.box.angles = array
+    vecs = np.asarray(box_vectors, dtype=np.float)
+    if vecs.shape == (3,3):
+        pass
+    else:
+        vecs.reshape(3,3)
+
+    return _normalize_box(vecs)
+
+def _lengths_angles_to_vectors(lengths, angles):
+    #print(f'lengths: {lengths}, angles:{angles}')
+    (a, b, c) = lengths
+    (alpha, beta, gamma) = np.deg2rad(angles)
+    #print(a,b,c,alpha,beta,gamma)
+
+    a_vec = np.asarray([a,0.0,0.0],)
+
+    b_x = b * np.cos(gamma)
+    b_y = b * np.sin(gamma)
+    b_vec = np.asarray([b_x, b_y, 0.0],)
+
+    c_x = c * np.cos(beta)
+    c_cos_y_term = ((np.cos(alpha) - (np.cos(beta) * np.cos(gamma))) / np.sin(gamma))
+    c_y = c * c_cos_y_term
+    c_z = c * np.sqrt(1 - np.square(np.cos(beta)) - np.square(c_cos_y_term))
+    c_vec = np.asarray([c_x, c_y, c_z])
+    box_vectors = np.asarray((a_vec,b_vec,c_vec))
+    #print(box_vectors)
+    box_vectors.reshape(3,3)
+    _validate_box_vectors(box_vectors=box_vectors)
+    return box_vectors
+
+
+def _normalize_box(vectors):
+    """Align the box matrix into a right-handed coordinate frame.
+
+    NOTE: This assumes that the matrix is in a row-major format.
+
+    NOTE: Inspiration and logic are from the Glotzer group
+    package, Garnett; which is provided under a BSD 3-clause License.
+    For additional information, refer to the License file provided with
+    this package.
+    """
+    det = np.linalg.det(vectors)
+    #print(f'det is: {det}')
+    if np.any(np.allclose(det, 0.0,atol=1e-5)):
```

Comment:
I think `det` should always be a scalar here, unless you have to worry about inputs that are sets of vectors belonging to multiple boxes. In that case, you can use:
```suggestion
    if np.isclose(det, 0.0, atol=1e-5):
```

Reply:
You are absolutely correct! the `np.any(np.allclose())` incantation is just muscle memory at this point. Good call!